lib/elixir/lib/gen_event/stream.ex

```diff
@@ -56,7 +56,7 @@ defmodule GenEvent.Stream do
 end
 
 defimpl Enumerable, for: GenEvent.Stream do
-  def reduce(stream, acc, fun) do
+  def reduce(stream, acc, fun) when is_function(fun, 2) do
```

Comment:
I don't think that this ever ends up getting called by the user directly, so maybe we can drop this one?